auto_editor/Fix1003Rule.py

```diff
@@ -17,5 +18,105 @@ def change_type(self) -> ChangeTypeEnum:
+                warning_code = warning_code + now_code + "\n"
+                prefix, new_code = self.remove_function_info(warning_code)
+                new_code = prefix + new_code
+                print("new_code:",new_code)
+
+                self.replace_useless_multiple_line(warning_last_line, i, all_lines)
+                new_lineItem = LineItem(new_code)
+                new_lineItem.original_line = warning_last_line+1
+                all_lines[warning_last_line+1] = new_lineItem
+                self.test_print( all_lines, warning_last_line)
+                tmp_dict[file_path] = all_lines
+                break
+
+        project.paths_to_lines[file_path] = tmp_dict[file_path]
         return project
+
+    def strip_determin(self,first_time,lines):
+        if first_time == False:
+            now_code = lines.code.strip()
+        else:
+            now_code = lines.code
+
+        return now_code
+
+    def replace_useless_multiple_line(self,warning_last_line,i,all_lines):
+        for j in range(warning_last_line + 1, i + 1):
+            empty_lineItem_instance = LineItem("")
+            empty_lineItem_instance.original_line = j-1
+            all_lines[j] = empty_lineItem_instance
+
+    def test_print(self,all_lines,warning_last_line):
+        print("******Fix1003 Result****")
+        print("all_lines[",warning_last_line,"]:", all_lines[warning_last_line ].code)
+        print("all_lines[",warning_last_line+1,"]:", all_lines[warning_last_line + 1].code)
+        print("all_lines[",warning_last_line+2,"]:", all_lines[warning_last_line + 2].code)
+        print("all_lines[",warning_last_line+3,"]:", all_lines[warning_last_line + 3].code)
+
+    def remove_function_info(self, warning_code):
+        prefix = count_prefix(warning_code)
+        new_code = warning_code
+        if "=(" in warning_code:
+            new_code = self.replace_condition("=(", warning_code)
+        elif "= (" in warning_code:
+            new_code = self.replace_condition("= (", warning_code)
+        elif "((" in warning_code:
+            new_code = self.replace_condition("((", warning_code)
+
+        return prefix, new_code
+
+    def replace_condition(self, type, warning_code):
+        new_code = warning_code.split(type)
+        if type == "((":
+            merged_warning_code = merge_except_function(new_code)
+            new_code = merged_warning_code.replace(",0));", ";")
+            new_code = new_code.replace(", 0));", ";")
+        else:
+            merged_warning_code = merge_except_function(new_code)
+            new_code = merged_warning_code.replace(",0);", ";")
+            new_code = new_code.replace(", 0);", ";")
+
+        return new_code
+
+
+def count_prefix(new_code):
+    j, prefix = 0, ""
+    while j < len(new_code):
+        if new_code[j] == " ":
+            prefix = prefix + " "
+            j += 1
+        else:
+            break
+
+    return prefix
+
+def merge_except_function(new_code):
```

Comment:
Looks like this is not indented and not part of the class. Is this intentional?

Reply:
merge_except_function is used in replace_condition，not indented is because this merge the multiple warning code condition, in this condition, only the first line need indented, others are not.  
like:
 CUDA_SAFE_CALL(

    (c->constant = sycl::malloc_device<float>(num_clusters, q_ct1), 0));

Reply:
Oh, sorry I think my question was not clear. I mean: why is this line `def merge_except_function(new_code):` (and the ones below) not indented? If it's not indented like now, it does not belong to the Fix1003Rule class, but is an independent function. And if you indent it, it becomes part of the class, like all the other methods above it

Reply:
ooooh，understand !! my bad